File: AccuPay/Domain/TimeLogs/BreakTimeBracketHelper.vb
Imports AccuPay.Entity

Public Class BreakTimeBracketHelper

    Friend Shared Function GetBreakTimeDuration(
                            breakTimeBrackets As IList(Of BreakTimeBracket),
                            shiftDuration As Double) As Decimal

        breakTimeBrackets = breakTimeBrackets.
                                OrderBy(Function(b) b.ShiftDuration).
                                ThenBy(Function(b) b.BreakDuration).
                                ToList

        Dim lastBreakTimeBracket As New BreakTimeBracket

        For Each breakTimeBracket In breakTimeBrackets

            lastBreakTimeBracket = breakTimeBracket

            If breakTimeBracket.ShiftDuration >= shiftDuration Then

                Exit For

            End If

        Next

        If lastBreakTimeBracket Is Nothing Then

            Return 0

        Else

            Return lastBreakTimeBracket.BreakDuration

        End If

    End Function
End Class

File: Accupay.Data/Helpers/BreakTimeBracketHelper.cs
using AccuPay.Data.Entities;
using System.Collections.Generic;
using System.Linq;

namespace AccuPay.Data.Helpers
{
    public class BreakTimeBracketHelper
    {
        public static decimal GetBreakTimeDuration(IList<BreakTimeBracket> breakTimeBrackets,
                                                    double shiftDuration)
        {
            breakTimeBrackets = breakTimeBrackets.
                                    OrderBy(b => b.ShiftDuration).
                                    ThenBy(b => b.BreakDuration).
                                    ToList();

            var lastBreakTimeBracket = new BreakTimeBracket();

            foreach (var breakTimeBracket in breakTimeBrackets)
            {
                lastBreakTimeBracket = breakTimeBracket;

                if (breakTimeBracket.ShiftDuration >= (decimal)shiftDuration)
                    break;
            }

            if (lastBreakTimeBracket == null)
                return 0;
            else
                return lastBreakTimeBracket.BreakDuration;
        }
    }
}